File: Medium/55. Jump Game/Solution.cpp
class Solution {
public:
//checking right to left if the current position can reach the last index
    bool canJump(vector<int>& nums) {
        int minjump=0;
        for(int i=nums.size()-2; i>=0; i--) {
            minjump++;
            if(nums[i]>=minjump) 
                minjump=0;
        }
        return !minjump; 
    }
};

File: Medium/55. Jump Game/Solution.java
public class Solution {
    public boolean jumpPossible(int[] nums) {
        int maxReach = 0;

        for (int i = 0; i < nums.length; i++) {
            if (i > maxReach) {
                return false;
            }

            maxReach = Math.max(maxReach, i + nums[i]);

            if (maxReach >= nums.length - 1) {
                return true;
            }
        }

        return false;
    }

    public static void main(String[] args) {
        Solution jumpGame = new Solution();
        int[] nums1 = {3, 2, 1, 0, 4};
        int[] nums2 = {2, 3, 1, 1, 4};

        boolean result1 = jumpGame.jumpPossible(nums1);
        boolean result2 = jumpGame.jumpPossible(nums2);

        System.out.println("Can reach the last index (nums1): " + result1); // Output: false
        System.out.println("Can reach the last index (nums2): " + result2); // Output: true
    }
}
